services/api/src/resources/environment/harborSetup.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+/* ::
+import type MariaSQL from 'mariasql';
+*/
+
+const harborClient = require('../../clients/harborClient');
+const logger = require('../../logger');
+const R = require('ramda');
+const Sql = require('../env-variables/sql');
+const { isPatchEmpty, prepare, query, whereAnd } = require('../../util/db');
+
+const lagoonHarborRoute = R.compose(
+  R.defaultTo('http://172.17.0.1:8084'),
+  R.find(R.test(/harbor-nginx/)),
+  R.split(','),
+  R.propOr('', 'LAGOON_ROUTES'),
+)(process.env);
+
+const createHarborOperations = (sqlClient /* : MariaSQL */) => ({
+  addEnvironment: async (openshiftProjectName, environmentID) => {
```

Comment:
we create a new harbor project for each Lagoon Environment? I don't think that's necessary? a harbor Project per Lagoon Project should be enough?

Reply:
This was resolved by switching to a single harbor project per lagoon project.